### PR TITLE
🐛 Mobile | Fix inconsistent score and partial UI update when switching profiles

### PR DESCRIPTION
**Pull request by AntPolkanov**
> 1. What triggered this change? (PBI link, Email Subject, conversation + reason, etc)

✏️ #786 , #789 

> 2. What was changed?

✏️ Used Reactive paradigm
Refactored and simplified UserService

> 3. Did you do pair or mob programming?

✏️ No

https://github.com/SSWConsulting/SSW.Rewards.Mobile/assets/123032112/7d504c52-4964-4c5e-b28d-42c03bab2e44

**Figure:** The score is updated on multiple pages & Profile switching updates all relevant data

### CODE REVIEW

File: src/MobileUI/Constants.cs [hunk 1]
```diff
@@ -20,7 +20,7 @@ public static class Constants
     public const string AuthRedirectUrl = "msauth.com.ssw.consulting://auth";
 
 #if DEBUG
-    public const string AuthorityUri = "https://app-ssw-ident-staging-api.azurewebsites.net";
+    public const string AuthorityUri = "https://identity.ssw.com.au"; // "https://app-ssw-ident-staging-api.azurewebsites.net";
```
Comment (matt-goldman) [inline src/MobileUI/Constants.cs:23]: Should probably revert this before merging

File: src/MobileUI/PopupPages/LinkSocial.xaml.cs [hunk 2]
```diff
@@ -20,13 +20,13 @@ public LinkSocial(Achievement achievement)
 
         private async void SaveButton_Clicked(object sender, System.EventArgs e)
         {
-            var msg = new SocialUsernameAddedMessage(new SocialAchievement
```
Comment (matt-goldman) [inline src/MobileUI/PopupPages/LinkSocial.xaml.cs:23]: Delete these

File: src/MobileUI/ViewModels/ProfileViewModels/MyProfileViewModel.cs [hunk 3]
```diff
@@ -1,106 +1,51 @@
...
-    ISnackbarService snackbarService,
     ILeaderService leaderService,
     IDevService devService,
     IPermissionsService permissionsService)
-    : ProfileViewModelBase(rewardsService, userService, snackbarService, devService, permissionsService),
-        IRecipient<ProfilePicUpdatedMessage>,
-        IRecipient<PointsAwardedMessage>,
-        IRecipient<SocialUsernameAddedMessage>
+    : ProfileViewModelBase(rewardsService, userService, devService, permissionsService),
+        IRecipient<PointsAwardedMessage>
```
Comment (matt-goldman) [inline src/MobileUI/ViewModels/ProfileViewModels/MyProfileViewModel.cs:14]: Should this still be implementing this `IRecipient`? I think the points change is included in this PR isn't it?

**Review by matt-goldman — CHANGES_REQUESTED** [review]
Few minor things to clear up, see comments.

One major one - we shouldn't be exposing `BehaviorSubject<T>`, instead this should be private, and we should expose `IObservable<T>()`. There are a few reasons for this, including better alignment with Rx principles, easier testing, but most importantly encapsulation. If you expose the behavior subject, a consumer can call `OnNext` on it; with `IObservable` they can't.

What are your thoughts on my `State<T>` approach? We don't necessarily need this, just wondering whether you think it's a good fit.

But definitely need to change to not exposing the behavior subject in the interface.

File: src/MobileUI/PopupPages/LinkSocial.xaml.cs (hunk 2 above)
Comment (AntPolkanov) [reply src/MobileUI/PopupPages/LinkSocial.xaml.cs:23]: Removed the entire file

File: src/MobileUI/Constants.cs (hunk 1 above)
Comment (AntPolkanov) [reply src/MobileUI/Constants.cs]: Yeah... but now in 95% of cases we use production Identity Server in Debug. I added another line which points to staging and commented it out. 
Is it ok with you?

File: src/MobileUI/ViewModels/ProfileViewModels/MyProfileViewModel.cs (hunk 3 above)
Comment (AntPolkanov) [reply src/MobileUI/ViewModels/ProfileViewModels/MyProfileViewModel.cs]: Yes, now we can remove this message. Thanks.

**Comment by AntPolkanov** [comment]
Thanks @matt-goldman. The PR is ready for another review round.

> What are your thoughts on my State<T> approach? We don't necessarily need this, just wondering whether you think it's a good fit.

I haven't fully grasped it yet, but at first glance, it seems like we can benefit from this approach.

**Review by matt-goldman — APPROVED** [review]
LGTM

**Pull request merged**